tests/typecheck/fields/test_related.yml

```diff
@@ -608,8 +608,12 @@
         reveal_type(Article().registered_by_user)  # N: Revealed type is "myapp.models.MyUser*"
 
         user = MyUser()
-        reveal_type(user.book_set) # N: Revealed type is "django.db.models.manager.RelatedManager[myapp.models.Book]"
-        reveal_type(user.article_set) # N: Revealed type is "django.db.models.manager.RelatedManager[myapp.models.Article]"
+        reveal_type(user.book_set) # N: Revealed type is "myapp.models.MyUser_Book_RelatedManager1"
+        reveal_type(user.article_set) # N: Revealed type is "myapp.models.MyUser_Article_RelatedManager1"
+        reveal_type(user.book_set.add)  # N: Revealed type is "def (*objs: Union[myapp.models.Book*, builtins.int], *, bulk: builtins.bool =)"
+        reveal_type(user.article_set.add)  # N: Revealed type is "def (*objs: Union[myapp.models.Article*, builtins.int], *, bulk: builtins.bool =)"
+        reveal_type(user.book_set.queryset_method())  # N: Revealed type is "builtins.int"
+        reveal_type(user.article_set.queryset_method())  # N: Revealed type is "builtins.int"
```

Comment:
I've extended this test to also check in on methods only existing on the custom queryset class. Basically to ensure the related manager is built from correct bases. I realised the bases were previously incorrect, so the related manager was built from `models.Manager` and `RelatedManager` which yielded an `attribute does not exist` error for any methods existing on the custom queryset.

I'm thinking that the asserts now should ensure a correct hierarchy. Perhaps we should also check on e.g. `user.book_set.filter` to see that the `Manager`/`BaseManager` is also a base and correct? Since now we're checking that methods from `RelatedManager` and `LibraryEntityQuerySet` are included

Reply:
> I'm thinking that the asserts now should ensure a correct hierarchy. Perhaps we should also check on e.g. user.book_set.filter to see that the Manager/BaseManager is also a base and correct?

Sure, why not?